ansible_runner/display_callback/module.py

```diff
@@ -21,18 +21,35 @@
 import collections
 import contextlib
 import datetime
+import os
 import sys
 import uuid
 from copy import copy
 
 # Ansible
 from ansible import constants as C
 from ansible.plugins.callback import CallbackBase
-from ansible.plugins.callback.default import CallbackModule as DefaultCallbackModule
+from ansible.plugins.loader import callback_loader
 
-# AWX Display Callback
 from .events import event_context
-from .minimal import CallbackModule as MinimalCallbackModule
+
+user_specified_callback = os.getenv('ORIGINAL_STDOUT_CALLBACK')
+runner_callback = os.environ.pop('ANSIBLE_STDOUT_CALLBACK', 'awx_display')
+
+if user_specified_callback:
+    os.environ['ANSIBLE_STDOUT_CALLBACK'] = user_specified_callback
+
+is_adhoc = os.getenv('AD_HOC_COMMAND_ID', False)
+
+# Dynamically construct base classes for our callback module, to support custom stdout callbacks.
+if is_adhoc:
+    default_stdout_callback = 'minimal'
+else:
+    default_stdout_callback = C.config.get_config_value('DEFAULT_STDOUT_CALLBACK')
```

Comment:
You're going through this process that modifies `os.environ`, which can have unforeseen side-effects for the user later, and I just don't see any need for it.

```python
>>> os.environ['ANSIBLE_STDOUT_CALLBACK'] = 'foo'
>>> C.config.get_config_value('DEFAULT_STDOUT_CALLBACK')
'foo'
```

So you have this value that the user set, you read it, you set the callback to it in `os.environ`, and then you read it back via `C.config.get_config_value`.

Instead, couldn't you just do this?

```python
if is_adhoc:
    default_stdout_callback = 'minimal'
else:
    default_stdout_callback = user_specified_callback
```

And drop the part about `os.environ['ANSIBLE_STDOUT_CALLBACK'] = user_specified_callback`